### PR TITLE
Handling situation when dataset which was previosly scheduled for bac…

### PR DESCRIPTION
**Pull request by radkomateusz**
…kup doesn't exist for the moment of time when task is executed. Now it doesn't throw an exception.

### CODE REVIEW

File: src/big_query/big_query.py [hunk 1]
```diff
@@ -72,7 +72,16 @@ def list_table_ids(self, project_id, dataset_id):
             projectId=project_id, datasetId=dataset_id
         )
         while request is not None:
-            tables = request.execute()
+            try:
+                tables = request.execute()
+            except HttpError as ex:
+                if ex.resp.status == 404:
```
Comment (przemyslaw-jasinski) [inline src/big_query/big_query.py]: Does service return 404 only for missing dataset?

File: src/big_query/big_query.py [hunk 2]
```diff
@@ -72,7 +72,16 @@ def list_table_ids(self, project_id, dataset_id):
...
+                    logging.info("Dataset not '%s:%s' Not Found", project_id,
+                               dataset_id)
+                    raise StopIteration
```
Comment (przemyslaw-jasinski) [inline src/big_query/big_query.py]: Is it recommended way in python? Couldn't we just give 'return' here?

File: src/big_query/big_query.py (hunk 1 above)
Comment (radkomateusz) [reply src/big_query/big_query.py]: no. I added additional validation for that situation

File: src/big_query/big_query.py (hunk 2 above)
Comment (radkomateusz) [reply src/big_query/big_query.py]: Return is better. Done.

File: src/big_query/big_query.py [hunk 3]
```diff
@@ -72,7 +72,16 @@ def list_table_ids(self, project_id, dataset_id):
             projectId=project_id, datasetId=dataset_id
         )
         while request is not None:
-            tables = request.execute()
+            try:
+                tables = request.execute()
+            except HttpError as ex:
+                if ex.resp.status == 404 and 'Not found: Dataset' in ex.content:
+                    logging.info("Dataset not '%s:%s' Not Found", project_id,
```
Comment (MZatorski) [inline src/big_query/big_query.py]: Duplicated 'not' in log message

File: tests/big_query/test_big_query.py [hunk 4]
```diff
@@ -167,6 +177,16 @@ def __create_tables_list_responses():
              content('tests/json_samples/bigquery_table_list_page_last.json'))
         ])
 
+    @staticmethod
+    def __create_tables_list_404_responses():
```
Comment (MZatorski) [inline tests/big_query/test_big_query.py]: The name of this method is confusing. Maybe just call it '__create_dataset_not_found_response'

File: src/big_query/big_query.py (hunk 3 above)
Comment (radkomateusz) [reply src/big_query/big_query.py]: done

**Pull request merged**